nltk/sem/relextract.py

```diff
@@ -198,17 +198,20 @@ def extract_rels(subjclass, objclass, doc, corpus='ace', pattern=None, window=10
     :return: see ``mk_reldicts``
     :rtype: list(defaultdict)
     """
-
-    if subjclass and subjclass not in NE_CLASSES[corpus]:
-        if _expand(subjclass) in NE_CLASSES[corpus]:
-            subjclass = _expand(subjclass)
-        else:
-            raise ValueError("your value for the subject type has not been recognized: %s" % subjclass)
-    if objclass and objclass not in NE_CLASSES[corpus]:
-        if _expand(objclass) in NE_CLASSES[corpus]:
-            objclass = _expand(objclass)
-        else:
-            raise ValueError("your value for the object type has not been recognized: %s" % objclass)
+    
+    for i, s in enumerate(subjclass):
```

Comment:
What if `subjclass` is a set, instead of a list. Then containment check in line 225 is sublienear.

Maybe, instead of modifying the passed `subjclass` it makes sense to build a new:
```python
#  Calculate _expand() for every subjclass, but also keep the original
expanded_subjclass = (
  (_expand(s), s)
  for s in subjclass
  if s and s not in NE_CLASSES[corpus]
)
#  Select between an expanded and the original subjclass
subjclass = set(
  e if _expand(s) in NE_CLASSES[corpus] else s
  for e, s in expanded_subjclass
)
# BUT WE STILL NEED TO RAISE AN EXCEPTION!
```